src/features/expenses/AddExpense/AddParticipants.tsx

```diff
@@ -0,0 +1,161 @@
+    return SplitMethod.Percentage;
+  else if (splitMethodIndex === 3) // Ajustment uses Equal as base type and AssignedAmount for adjustments
+    return SplitMethod.Equal;
+  else
+    throw new Error('Invalid split method index');
+};
+   
+
+const AddParticipants = () => {
+  const { t } = useTranslation('expenses');
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [createTransaction, { data: createdTransaction, isLoading: isTransactionCreating }] = useCreateTransactionMutation();
+
+  const transaction = useSelector(newTransactionSelector);
+  const group = useGetTransactionGroupByIdQuery(transaction?.groupId || '');
+
+  const [splitMethodIndex, setSplitMethodIndex] = useState<number>(0);
+
+  const handleBackAction = useCallback(() => {
+    navigate(-1);
+  }, [navigate]);
+
+  const resetPartialsAssignments = useCallback((splitIndex: number) => {
+    const newSplitMethod = splitMethodIndexToEnum(splitIndex);
+    
+    // Get unique usersIds from partialsAssignments
+    const userIds = transaction!.partialsAssignments
+      .map(partialsAssignment => partialsAssignment.userId)
+      .filter((userId, index, self) => index === self.indexOf(userId));
+
+    const newTransaction = {
+      ...transaction!,
+      partialsAssignments: [...userIds].map((userId) => ({
+        userId,
+        partialAmount: 0,
+        splitMethod: newSplitMethod
+      }))
+    };
+
+    dispatch(saveTransaction(newTransaction));
+  }, [transaction, dispatch]);
+
+  const onSplitMethods = useMemo(() => new Array(4).fill(null).map((_, i) => () => {
+    resetPartialsAssignments(i);
+    setSplitMethodIndex(i);
+  }), [setSplitMethodIndex, resetPartialsAssignments]);
+
+  const onSave = useCallback(() => {
+    createTransaction(transaction!);
+  }, [createTransaction, transaction]);
+
+  useEffect(() => {
+    if (createdTransaction) {
+      toast.success(t('transactionCreated'));
+      navigate(composeGroupRoute(transaction!.groupId), { replace: true });
+    }
+  }, [createdTransaction, navigate, t, transaction]);
+
+  return (
+    <PageWrapper>
+      <Header
+        text={t('participants')}
+        leftActionComponent={<BackIconSVG onClick={handleBackAction} />}
+      />
+      <ContentWrapper fullWidth>
+        <AddParticipantsWrapper fullWidth gap="5vh">
+          <ButtonBase>
+            <TextUnderline reversedColor>
+              {`${group.data?.participants.length}/${group.data?.participants.length} ${t('membersInTransaction')}`}
+            </TextUnderline>
+          </ButtonBase>
+
+          <SplitMethodWrapper fullWidth jc="space-between">
+            <SplitMethodBorderWrapper>
+              <BorderShift flex={splitMethodIndex} />
+              <MovingBorder />
+              <BorderShift flex={3 - splitMethodIndex} />
+            </SplitMethodBorderWrapper>
+
+            <SplitMethodButton text={t("equally")} svgImg={<EquallyMethodSVG />} onClick={onSplitMethods[0]} />
```

Comment:
Please try to use enum here

Reply:
Done in 81508842d73fd52a080d5628b9689b50a2437d2d